doc: release notes: Update contents for 2.2.0

jhedberg opened the pull request:
Add updated issue list, update conf.py with the new version, and
update the VERSION file

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

jhedberg commented:
To be merged after #23357

carlescufi reviewed (commented):
Why did you change permissions on the release notes rst file?

jhedberg commented:
> Why did you change permissions on the release notes rst file?

That was unintentional. Probably the result of a shell redirect of the issue list into the file. Will fix.

jhedberg commented:
>> Why did you change permissions on the release notes rst file?

> That was unintentional. Probably the result of a shell redirect of the issue list into the file. Will fix. 

Actually, I didn't change the permissions. Why do you say I did?

ioannisg commented on VERSION:
```diff
@@ -2,4 +2,4 @@ VERSION_MAJOR = 2
 PATCHLEVEL = 0
 VERSION_TWEAK = 0
-EXTRAVERSION = rc3
+EXTRAVERSION =
```
I tihnk you need a separate PR for this change.

jhedberg replied:
I was just  following what what was done for 2.1 with commit b413223a66ed9e857c67ee21d1e8be4768804dd1

jhedberg replied:
Anyway, why would a separate PR be needed?

ioannisg replied:
Well, actually, a separate commit (but then a separate PR would also make CI run once more).
You should update the version file with a commit with "release: " in the title, and appropriate content.
https://github.com/zephyrproject-rtos/zephyr/wiki/Release-procedure

ioannisg commented on doc/conf.py:
```diff
@@ -527,6 +527,7 @@
     'theme_logo_only': False,
     'current_version': version,
     'versions': (("latest", "/"),
+                 ("2.2.0", "/2.2.0/"),
```
can you also remove one from the bottom here, so we got to display the same number of releases?

jhedberg replied:
sure

jhedberg replied:
actually, I could remove two versions since 2.1 didn't remove any

jhedberg replied:
or perhaps even three, so we start the list with the LTS release?

ioannisg replied:
that's also fine with me.

Commit:
doc: release notes: Update contents for 2.2.0

Finalize the release notes for Zephyr 2.2.0.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
doc: Update generated version list for 2.2.0

Add 2.2.0 to the version pick list. Also remove old versions so that
the oldest one is the LTS release.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
release: Zephyr 2.2.0

Bump version to 2.2.0 final.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

jhedberg commented:
@ioannisg I've now split the PR into three commits, one for each file and with the last commit updating `VERSION`. I also removed all 1.x versions from `doc/conf.py` except for the last one, i.e. the LTS release.
